### PR TITLE
SETTINGS: Validation for personal data

### PR DESCRIPTION
**Pull request by eunjuhuss**
#### Description:
We have some issues associated to validation for personal data. 
We decide to allow the user to put anything they want (include emojis, special chars, numbers) in the input. 
The reason is that some countries allows the use of number and special chars in name.
The use of emojis in name is allowed due to the fact that we can't find a reliable way of ensuring that all emojis unicode can be blocked without causing issues with the validation.

This PR is rectifying the below issues.
Issue 472 https://github.com/SUNET/eduid-front/issues/472 (Not possible to have "-" in your name)
Issue 475 https://github.com/SUNET/eduid-front/issues/475 (Validation acting strangely)
Issue 476 https://github.com/SUNET/eduid-front/issues/476 (Text on button)

- Button text changed to `SAVE/SPARA`
- Added validation checking for white spaces and trim function for sending trimmed data to backed when the user is saving there name with white spaces infront or after the name.
- An error will displayed when the user has not update the input value, this will inform the user why the save button is disabled.
- Removed unused translate texts.
- Added  `touchOnChange: true, touchOnBlur: true` in PdataForm to get active error messages
- Removed `keepDirtyOnReinitialize: true`in PdataForm, I found a problem with the language selected buttons, if changed and then reverted the save button was still active, now it will inform the user that the value is unchanged and the button will remain disabled

**Good to know**: After I added trim function in sagas, unit test failed, that's why I commented out test the code in `lookUpMobile-test.js` this will be fixed in a separate branch. 

----
- Before and After 
_when user write their name in the input, earlier the user got a feedback first on "not focusing", now the feedback will show even when active, green border line and checked icon_

Screenshot 2021-01-19 at 11 53 23

- Before and After 
_when the user has not update the input value, this will inform the user why the save button is disabled._
Screenshot 2021-01-19 at 11 53 31

----

#### For reviewer:
- [ ] Read the above description
- [ ] Reviewed the code changes
- [ ] Navigate to the branch (pulled the latest version)
- [ ] Run the code and been able to execute the expected function
- [ ] Check any styling on both desktop and mobile sizes

### CODE REVIEW

File: src/login/components/Inputs/CustomInput.js [hunk 1]
```diff
@@ -56,7 +56,7 @@ const RenderInput = props => {
         <Fragment key={index}>
           <label key={option[0]} htmlFor={option[1]}>
           <input
-            className={"radio-input"}
+            className={props.meta.error && props.meta.visited ? "radio-input error":"radio-input"}
```
Comment (njons) [inline src/login/components/Inputs/CustomInput.js:59]: I am not sure this condition is working. Upon error, I am able to see the text informing me that "the value is not changed", but I am not able to see a change in class or a change in colour inside the radio button.

File: src/login/styles/_inputs.scss [hunk 2]
```diff
@@ -93,15 +97,15 @@ input::placeholder {
...
+.form-control:focus,
 input:focus {
   outline: 0;
   border-bottom: solid 2px $black;
   box-shadow: 0 0 0 0.2rem $mid-gray;
 }
 
 .form-control.is-valid:focus {
-  border-color: $black;
+  border-color: #28a745;
```
Comment (njons) [inline src/login/styles/_inputs.scss:108]: Should this colour be turned into a variable?

File: src/sagas/PersonalData.js [hunk 3]
```diff
@@ -103,7 +103,18 @@ export function fetchAllPersonalData(config) {
...
+      .querySelector("input").value.trim(),
+    surname: document
+      .getElementById("surname")
+      .querySelector("input").value.trim(),
+    display_name: document
+      .getElementById("display_name")
+      .querySelector("input").value.trim(),
+    language: document
+      .getElementById("language")
+      .querySelector('input[type="radio"]:checked').value,
```
Comment (njons) [inline src/sagas/PersonalData.js:117]: Is it be possible to make this less dependent on the html? If not we should leave it, but I thought I'd ask as we might change the html in the future.

File: src/components/PersonalData.js [hunk 4]
```diff
@@ -2,37 +2,22 @@ import React, { Component } from "react";
...
-
-  ["given_name", "surname", "display_name", "language"].forEach((pdata) => {
-    if (!values[pdata]) {
-      errors[pdata] = "required";
-    }
-    //add space pattern instead of trim because the input datas will be added without space.
-    else if(spacePattern.test(values[pdata])){
-      errors[pdata] = "required";
+  ["given_name", "surname", "display_name", "language"].forEach((inputName) => {
+    if (!values[inputName] || !values[inputName].trim()) {
```
Comment (njons) [inline src/components/PersonalData.js:15]: When I add spaces before or after a saved value I expect to get an error (and to see the "required" text), but the input is green and I am able to save the value. Have I misunderstood this code?

File: src/sagas/PersonalData.js (hunk 3 above)
Comment (eunjuhuss) [reply src/sagas/PersonalData.js]: yes! I have updated!

File: src/login/styles/_inputs.scss (hunk 2 above)
Comment (eunjuhuss) [reply src/login/styles/_inputs.scss]: this has been updated after merge from latest mater

File: src/login/components/Inputs/CustomInput.js (hunk 1 above)
Comment (eunjuhuss) [reply src/login/components/Inputs/CustomInput.js:59]: I have tried and it is working in my environment, can you please check again.

File: src/components/PersonalData.js (hunk 4 above)
Comment (eunjuhuss) [reply src/components/PersonalData.js]: I have updated (using empty string regex pattern)

**Comment by eunjuhuss** [comment]
-Removed "props.invalid" in the button and created function to set "state" isDisable.
-Use useEffect to get initial personal data to match updated data if all the updated values are matched with initial values, button will be disabled, otherwise abled.
-Added "onChange" function for setState updating personal data.

**Pull request merged**